[MIG] product_order_noname: Migration to 15.0

nicomacr opened the pull request.

Commit:
[ADD] product_order_noname: Speedup product retrieve

This module improves retrieving products when a language is installed
and configured. The default behaviour is to order products by name,
which in this case would be slow because Odoo will process the
product's name to translate it.

When ordering products by name and having a language configured the
Odoo's ORM makes a join to `product_template` an then another join but
this time with a nested SELECT to retrieve the translation. This is very
slow because the lack of index on the JOIN, making a query like
`env['product.product'].search([], limit=10)` taking up to 1027.282 ms.

With the changes introduced here the query is considerably simpler, just
as the following:

    SELECT "product_product".id
    FROM   "product_product"
    WHERE  ( "product_product"."active" = true )
    ORDER  BY "product_product"."default_code"
    LIMIT  10

Also the execution time dramatically decrease, from 1027.282 ms to
0.624 ms which is about 1.65k times faster.

moylop260 committed:
[REF] product_order_noname: Change product.template order

Product.template has `_order='name'`
even if the `name` field has an `index=True`
This index is not used because of this field has a
`translate=True` parameter

So, it is in the same situation like product.product

So, it is changed using `_order='default_code, id'` instead.
For a faster sort it requires an index so the field `default_code`
is redefined to `index=True`

OCA-git-bot committed:
product_order_noname 12.0.2.0.0

oca-travis committed:
[UPD] Update product_order_noname.pot

OCA-git-bot committed:
[ADD] icon.png

Yvesldff committed:
Added translation using Weblate (French)

Yvesldff committed:
Translated using Weblate (French)

Currently translated at 83.3% (5 of 6 strings)

Translation: product-attribute-14.0/product-attribute-14.0-product_order_noname
Translate-URL: https://translation.odoo-community.org/projects/product-attribute-14-0/product-attribute-14-0-product_order_noname/fr/

nicomacr committed:
[IMP] product_order_noname: black, isort, prettier

nicomacr committed:
[MIG] product_order_noname: Migration to 15.0

florenciafrigieri2 reviewed (approved):
LGTM

jjscarafia commented on product_order_noname/models/product.py:
```diff
@@ -0,0 +1,6 @@
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+    _order = "default_code, id"
```
please use new order criteria
    _order = 'priority desc, default_code, id'

https://github.com/odoo/odoo/blob/13.0/addons/product/models/product.py#L70

jjscarafia commented on product_order_noname/models/product_template.py:
```diff
@@ -0,0 +1,8 @@
+
+class ProductTemplate(models.Model):
+    _inherit = "product.template"
+    _order = "default_code, id"
```
please use [new order](https://github.com/odoo/odoo/blob/15.0/addons/product/models/product_template.py#L19) criteria
    _order = "priority desc"

may be, I would add "default_code" here because if not the order will be only regarding the priority field

So, I would leave     _order = "priority desc, default_code"

jjscarafia reviewed (approved):
LGTM

nicomacr commented:
@legalsylvain could you check this PR?
 Thanks

florenciafrigieri2 commented:
@legalsylvain @rousseldenis Hi! Can you please check this PR? Thanks!

legalsylvain reviewed (approved):
Please add a section in the roadmap that mention to check if the module is still interesting in V16. (Due to ir.translation refactoring.)
Otherwise lgtm. Thanks.

nicomacr commented:
> Please add a section in the roadmap that mention to check if the module is still interesting in V16. (Due to ir.translation refactoring.)
> Otherwise lgtm. Thanks.

I've added this and updated the commit!

dreispt commented:
/ocabot merge nobump
